### PR TITLE
Docker-compose with MongoDB

### PR DESCRIPTION
**Pull request by amper43**
## Changes
- `analytics/Dockerfile` - removed unnecessary apt-get commands, remove unnecessary volume
-  `server/Dockerfile` - multi-stage build
- `docker-compose-mongo.yml` - new deployment with mongodb: 3 services:
   * server
   * analytics
   * mongo
- `HASTIC_API_KEY ` and `GRAFANA_URL` variables are made required using `${VAR_NAME?"error string"}` syntax. These vars should be set in environment before `docker-compose run`.
- add `mongo-init.js` startup script for MongoDB. It's used to init the DB in `docker-compose-mongo.yml`

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -27,9 +27,6 @@ services:
     build: analytics
     restart: always
     command: ${HASTIC_ANALYTICS_COMMAND}
-    #set HASTIC_ANALYTICS_COMMAND="python -um ptvsd --host 0.0.0.0 --port 5678 bin/server" for remote debug
```
Comment (rozetko) [inline docker-compose.yml]: is this comment not actual?

Comment (amper43) [reply docker-compose.yml]: removed

File: docker-compose-mongo.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    image: hastic/server:latest
+    build:
+      dockerfile: server/Dockerfile
+      context: .
+    environment:
+      ZMQ_CONNECTION_STRING: ${ZMQ_CONNECTION_STRING:-tcp://analytics:8002}
+      HASTIC_API_KEY: ${HASTIC_API_KEY?"HASTIC_API_KEY variable required"}
+      HASTIC_WEBHOOK_URL: ${HASTIC_WEBHOOK_URL}
+      HASTIC_WEBHOOK_TYPE: ${HASTIC_WEBHOOK_TYPE}
+      GRAFANA_URL: ${GRAFANA_URL?"GRAFANA_URL variable required"}
```
Comment (rozetko) [inline docker-compose-mongo.yml]: - let's move required variables to the top of  `environment`
- `... variable required` -> `... variable is required`

File: docker-compose-mongo.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
+version: '2.1'
+services:
+  server:
+    image: hastic/server:latest
+    build:
+      dockerfile: server/Dockerfile
+      context: .
+    environment:
+      ZMQ_CONNECTION_STRING: ${ZMQ_CONNECTION_STRING:-tcp://analytics:8002}
+      HASTIC_API_KEY: ${HASTIC_API_KEY?"HASTIC_API_KEY variable required"}
```
Comment (rozetko) [inline docker-compose-mongo.yml]: same here

File: docker-compose.yml [hunk 4]
```diff
@@ -7,10 +7,10 @@ services:
       context: .
     environment:
       ZMQ_CONNECTION_STRING: ${ZMQ_CONNECTION_STRING:-tcp://analytics:8002}
-      HASTIC_API_KEY: ${HASTIC_API_KEY}
+      HASTIC_API_KEY: ${HASTIC_API_KEY?"HASTIC_API_KEY variable required"}
```
Comment (rozetko) [inline docker-compose.yml]: same here

File: mongo-init.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+db.createUser(
+    {
```
Comment (rozetko) [inline mongo-init.js]: We use 2 spaces :)

File: server/Dockerfile [hunk 6]
```diff
@@ -1,23 +1,32 @@
...
 
-RUN apt-get install curl \
-                    bash \
-                    gnupg \
-                    make \
-                    g++ \
- && curl -sL https://deb.nodesource.com/setup_8.x | bash - \
- && apt-get update \
- && apt-get install nodejs
+#tools for building zeromq
```
Comment (rozetko) [inline server/Dockerfile]: let's add a space after `#`

File: docker-compose-mongo.yml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+  mongo:
+    image: mongo:4.2.1-bionic
+    environment:
+      MONGO_INITDB_ROOT_USERNAME: ${HASTIC_MONGO_ADMIN_USER:-admin}
+      MONGO_INITDB_ROOT_PASSWORD: ${HASTIC_MONGO_ADMIN_PASSWORD:-password}
+      MONGO_INITDB_DATABASE: ${HASTIC_MONGO_DB:-hastic}
+    restart: always
+    ports:
+      - 27017:27017
```
Comment (rozetko) [inline docker-compose-mongo.yml]: Do we need to publish the port? What if there is already a mongodb instance on host?

Comment (amper43) [reply docker-compose-mongo.yml]: I think user can add port publishing if it required
removed, thx

**Pull request merged**